rhp/v4/rpc_test.go

```diff
@@ -1276,7 +1276,7 @@ func TestRPCTimeout(t *testing.T) {
 
 	isTimeoutErr := func(err error) bool {
 		t.Helper()
-		return err != nil && (strings.Contains(err.Error(), "deadline exceeded") || strings.Contains(err.Error(), "i/o timeout"))
+		return err != nil && (strings.Contains(err.Error(), "deadline exceeded") || strings.Contains(err.Error(), "i/o timeout") || strings.Contains(err.Error(), "canceled by local with error code 1"))
```

Comment:
This is annoying. We keep accumulating error that might happen when we reach a timeout. So we must not forget to update `shouldResetTransport`. This error is effectively the "stream was gracefully closed" error of QUIC.